### PR TITLE
omniorb: add python3 to bindings and dependencies

### LINKED ISSUE #46586: omniorb only build a python2 package, nothing for python3
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [n/a] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [n/a] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

Install the `omniorb` package.

## What happened (include command output)

`omniorb` should build a python package, but only builds one for python 2.7.

## What you expected to happen

`omniorb` has supported python 3 for years.  Homebrew's default python version is 3.  Python 2 will be EOL'd in ≈1 month.  Homebrew shouldn't only ship python 2.7 packages.

## Step-by-step reproduction instructions (by running `brew install` commands)

`brew install omniorb`

### PR DESCRIPTION
**Pull request by alexchandel**
Fixes #46586

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/omniorb.rb [hunk 1]
```diff
@@ -19,6 +19,7 @@ class Omniorb < Formula
   end
 
   depends_on "pkg-config" => :build
+  depends_on "python"
```
Comment (carlocab) [inline Formula/omniorb.rb:22]:
```suggestion
  depends_on "python@3.9"
```
The formula is called `python@3.9` now.

**Review by carlocab — COMMENTED** [review]
Thanks for your PR, @alexchandel.

**Comment by carlocab** [comment]
Please also add a `revision 1` line after the `license`, so that users who have this installed get the updated version of the formula. Thanks!

File: Formula/omniorb.rb (hunk 1 above)
Comment (alexchandel) [reply Formula/omniorb.rb:22]: Must it be tied to a specific version?  Its bindings work with any version of Python 3.

I remember how long it took to update to Python 3.8, I don't want this to get in the way of the next update unless it's necessary.

Comment (SMillerDev) [reply Formula/omniorb.rb]: `depends_on "python"` means the same thing, just implicitly. The slow migration to python 3.8 is actually the reason this needs to be specified like this. It allows us to move the `python` alias without breaking everything that depends on "any python"

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**